src/linker/pipeline.py

```diff
@@ -153,3 +153,15 @@ def _validate_implementations(self) -> bool:
             return False
         else:
             return True
+
+    def _validate_input_data(self) -> bool:
```

Comment:
Oh good, glad you quickly figured out this convoluted mess of error message handling